src/sidebar/media-embedder.js

```diff
@@ -100,6 +101,29 @@ var embedGenerators = [
     return null;
   },
 
+/** Match Internet Archive URLs
+ *  The patterns are:
+ *  1. https://archive.org/embed/WHDH_20151121_043400_The_Tonight_Show_Starring_Jimmy_Fallon?start=360&end=420.3
+ *  2. https://archive.org/details/WHDH_20151121_043400_The_Tonight_Show_Starring_Jimmy_Fallon/start/360/end/420.3
+ *  The patterns are invariant, start and stop are always present.
+ */
+  function iFrameFromInternetArchiveLink(link) {
+    if (link.hostname !== 'archive.org') {
+      return null;
+    }
+
+    var groups = /(embed|details)\/(.+)$/.exec(link.href);
```

Comment:
The regex here ought to start with `^\/` to ensure that it only matches paths beginning with `/embed`or `/details`. Otherwise it could match other URLs that happen to contain "embed" or "details" somewhere else in the URL.